### PR TITLE
Refactor nodeShouldRunDaemonPod

### PR DESCRIPTION
**Pull request by notpad**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

After #82795 got merged, `nodeShouldRunDaemonPod` always returns the same value in `wantToRun` and `shouldSchedule`. So we can remove the `wantToRun` return value. And the `Predicates` function only calls `checkNodeFitness`, so we can move the logic of `checkNodeFitness` into `Predicates` to avoid some duplicate code.

**Which issue(s) this PR fixes**:

Part of #82702

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig scheduling
/cc @k82cn @draveness @ahg-g

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test

Thanks for this

**Comment by draveness** [comment]
> And the Predicates function only calls checkNodeFitness, so we can move the logic of checkNodeFitness into Predicates to avoid some duplicate code.

Could you explain why we should remove the `checkNodeFitness`?

**Comment by notpad** [comment]
Before 'ScheduleDaemonSetPods' was upgraded to GA, we need to check the featuregate in Predicates function, so it has two branches, one for the new logic when feature is enabled, and the other branch for the old logic when feature is disabled. In this case, it is reasonable to extract the new logic into a single function checkNodeFitness.

Now that we have removed the old logic, you can see what the Predicates function does is only calling the checkNodeFitness function. And they have some common preprocessing/postprocessing logic which are duplicated. 
```
    var predicateFails []predicates.PredicateFailureReason

    fit, reasons, err := ...
    if err != nil {
        return false, predicateFails, err
    }
    if !fit {
        predicateFails = append(predicateFails, reasons...)
    }

    return len(predicateFails) == 0, predicateFails, nil
```

Basically, it's just a code refactor, not sure if you have other concerns.

**Comment by xiaoxubeii** [comment]
@notpad The codes are lgtm for me. But do we need to change the title or divide it into two PRs to match the submitted codes?

**Comment by notpad** [comment]
@xiaoxubeii I have changed the title, thanks.

**Comment by notpad** [comment]
@draveness Did you get a chance to take another look?

**Review by draveness — APPROVED** [review]
Generally looks good, Thanks

 BTW: I was planning to fix this after the code freeze, but thanks for the work anyway.

/lgtm

**Comment by notpad** [comment]
Thanks @draveness

**Comment by ahg-g** [comment]
Thanks, but this isn't a scheduler change, you need someone from sig-apps to approve.

/unacc

**Comment by notpad** [comment]
Needs an approve, can anyone from sig-apps help with this?  @kubernetes/sig-apps-pr-reviews

**Comment by notpad** [comment]
Hi @janetkuo, could you take a look at this PR, it is a code refactor? Thanks.

**Comment by draveness** [comment]
Thanks @notpad, I'm going to take over the refactor.

**Comment by notpad** [comment]
@draveness This was already done in #86730
